packages/components/src/ActionIntercom/Intercom.component.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useLayoutEffect, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { translate } from 'react-i18next';
+
+import TooltipTrigger from '../TooltipTrigger';
+import Icon from '../Icon';
+import IntercomService from './Intercom.service';
+import getDefaultT from '../translate';
+import I18N_DOMAIN_COMPONENTS from '../constants';
+
+import theme from './Intercom.scss';
+
+function Intercom({ id, className, config, t }) {
+	const [show, setShow] = useState(false);
+	const ref = useRef(null);
+
+	// init intercom messenger, attaching it to the custom button
+	useEffect(() => {
+		IntercomService.boot(`#${id}`, config);
+		IntercomService.onShow(setShow.bind(null, true));
+		IntercomService.onHide(setShow.bind(null, false));
+
+		return IntercomService.shutdown;
+	}, [config, setShow]);
```

Comment:
This is not my strongest area, but I'm trying to think when `setShow` would change? 
Would you not also want to pass `id` here?

Reply:
Same remark as @annilsson unless I'm missing something, extra params passed to `useEffect` seems to be useful for performance tweaks and avoid useless recomputation (a bit like `useMemo` does) (ref here https://reactjs.org/docs/hooks-effect.html @ "Tip: Optimiing Performance by Skipping Effects")

Reply:
It's possible that `setShow` doesn't change. But in this case, having it in the hook dependencies has no impact. In all cases, that makes the hook dependencies correct.
But agree with you, id is a missing dependency. I add it.